next.config.js

```diff
@@ -1,67 +1,84 @@
-const path = require('path');
-const nextMDX = require('@next/mdx');
-const rehypePrism = require('@mapbox/rehype-prism');
+const { join } = require('path');
+const withPlugins = require('next-compose-plugins');
+const withBundleAnalyzer = require('@next/bundle-analyzer');
+const withMDX = require('@next/mdx');
 const withOptimizedImages = require('next-optimized-images');
+const rehypePrism = require('@mapbox/rehype-prism');
 const admonitions = require('remark-admonitions');
-const withBundleAnalyzer = require('@next/bundle-analyzer')({
-  enabled: process.env.ANALYZE === 'true',
-});
 
-const withMDX = nextMDX({
-  extension: /\.mdx?$/,
-  options: {
-    remarkPlugins: [admonitions],
-    rehypePlugins: [rehypePrism],
+const nextConfig = {
+  experimental: {
+    optimizeFonts: true,
+    optimizeCss: true,
+    babelMultiThread: true,
   },
-});
+  target: 'experimental-serverless-trace',
+  rewrites: () => [
+    {
+      source: '/feed.xml',
+      destination: '/_next/static/feed.xml',
+    },
+    {
+      source: '/sitemap.xml',
+      destination: '/_next/static/sitemap.xml',
+    },
+  ],
+  webpack(config, { dev, isServer }) {
+    if (!dev && isServer) {
+      const originalEntry = config.entry;
 
-module.exports = withBundleAnalyzer(
-  withMDX(
-    withOptimizedImages({
-      experimental: {
-        optimizeFonts: true,
-        optimizeCss: true,
-        babelMultiThread: true,
-      },
-      target: 'experimental-serverless-trace',
-      optimizeImagesInDev: false,
-      handleImages: ['jpeg', 'jpg', 'png', 'svg'],
-      inlineImageLimit: 1000,
-      pageExtensions: ['tsx', 'md', 'mdx'],
-      rewrites: () => [
-        {
-          source: '/feed.xml',
-          destination: '/_next/static/feed.xml',
-        },
-        {
-          source: '/sitemap.xml',
-          destination: '/_next/static/sitemap.xml',
-        },
-      ],
-      webpack5: false,
-      webpack(config, { dev, isServer }) {
-        if (!dev && isServer) {
-          const originalEntry = config.entry;
+      config.entry = async () => {
+        const entries = { ...(await originalEntry()) };
+
+        entries['./lib/build.ts'] = './lib/build.ts';
```

Comment:
Everything looks fine, but deployment to Vercel fails with the following error
```text
HookWebpackError: File /vercel/path1/.next/server/lib/build.ts.js does not exist.
```
I think it related to this line

Reply:
Super weird, is it something with our setup that causes that?

Reply:
let's try without it, it's for RSS and sitemap.xml but we run this script with ts-node right after `next build`. Not sure why it's here